lib/_tls_wrap.js

```diff
@@ -383,13 +395,6 @@ TLSSocket.prototype._init = function(socket, wrap) {
   if (options.handshakeTimeout > 0)
     this.setTimeout(options.handshakeTimeout, this._handleTimeout);
 
-  // Socket already has some buffered data - emulate receiving it
-  if (socket && socket._readableState && socket._readableState.length) {
-    var buf;
-    while ((buf = socket.read()) !== null)
-      ssl.receive(buf);
-  }
-
```

Comment:
I moved it to put it in the `process.nextTick` that I added to the constructor. If we want to just do the `nextTick` here AND in the constructor around `read(0)`, I'm fine doing it in both places, just let me know.